src/components/AddFormControl.js

```diff
@@ -0,0 +1,256 @@
+      zipCode: '',
+      userId: '',
+      date_time_added: ''
+    };
+    this.listingChanged = this.listingChanged.bind(this);
+    this.requestChanged = this.requestChanged.bind(this);
+    this.captureData = this.captureData.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.submitHandler = this.submitHandler.bind(this);
+  }
+
+  captureData(inputType, data) {
+    alert(inputType);
+    switch (inputType) {
+      case 'title':
+        this.setState({ title: data });
+        break;
+    }
+  }
+
+  listingChanged() {
+    this.setState({
+      listingType: 'listing'
+    });
+  }
+
+  requestChanged() {
+    this.setState({
+      listingType: 'requesting'
+    });
+  }
+
+  submitHandler = event => {
+    event.preventDefault();
+    // alert('You are submitting');
+    console.log(event);
+
+    fetch(
+      'https://cors-anywhere.herokuapp.com/https://api-dot-pacific-plating-262123.appspot.com/api/v1.0/items/categories',
+      {
+        method: 'GET'
+      }
+    )
+      .then(response => {
+        return response.json();
+      })
+      .then(jsonData => {
+        console.log(jsonData);
+      });
+
+    fetch(
+      'https://cors-anywhere.herokuapp.com/https://api-dot-pacific-plating-262123.appspot.com/api/v1.0/items/postnew',
+      {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          listingType: this.state.listingType,
+          itemType: this.state.itemType,
+          title: this.state.title,
+          description: this.state.description,
+          price: this.state.price,
+          date_time_added: Date(),
+          test: 'yes'
+        })
+      }
+    );
+  };
+
+  handleInputChange(event) {
+    const target = event.target;
+    const value = target.value;
+    const id = target.id;
+    const name = target.name;
+    console.log('value: ' + value);
+    console.log('id: ' + id);
+    console.log('name: ' + name);
+
+    switch (id) {
+      case 'item-radio':
+        this.setState({ [name]: 'item' }, () => {
+          console.log(this.state);
+        });
+        break;
+      case 'time-radio':
+        this.setState({ [name]: 'time' }, () => {
+          console.log(this.state);
+        });
+        break;
+      case 'requesting-radio':
+        this.setState({ [name]: 'requesting' }, () => {
+          console.log(this.state);
+        });
+        break;
+      case 'listing-radio':
+        this.setState({ [name]: 'listing' }, () => {
+          console.log(this.state);
+        });
+        break;
+      case 'title':
+        this.setState({ [name]: 'test' }, () => {
+          console.log(this.state);
+        });
+        break;
+      default:
+        this.setState({ [name]: value }, () => {
+          console.log(this.state);
+        });
+        break;
+    }
+
+    // this.setState({
+    //   [name]: value
```

Comment:
seemed like you were on the right track here. I used a version of this on my branch, but ideally we'd have a different function for each component to prevent side effects.